collector/filesystem_linux.go

```diff
@@ -28,8 +30,12 @@ const (
 	defIgnoredMountPoints = "^/(dev|proc|sys|var/lib/docker)($|/)"
 	defIgnoredFSTypes     = "^(autofs|binfmt_misc|cgroup|configfs|debugfs|devpts|devtmpfs|fusectl|hugetlbfs|mqueue|overlay|proc|procfs|pstore|rpc_pipefs|securityfs|sysfs|tracefs)$"
 	readOnly              = 0x1 // ST_RDONLY
+	mountTimeout          = 30 * time.Second
 )
 
+var stuckMounts = make(map[string]struct{})
+var mutex = &sync.Mutex{}
```

Comment:
Rather than giving it a generic type name, name a mutex after what it is protecting, like `stuckMountsMtx`.